(maint) PR Reviewer checklist (#368)
puppetlabs/puppetlabs-testing: .github/pull_request_template.md

## Contributing

> As a contributor please ensure you have read over and understand Puppets

Reviewer: ```suggestion
As a contributor please ensure you have read over and understand Puppet's
```

> Contributing guidelines, they can be found in the root directory of all Supported Modules.

Reviewer: Following up on previous conversations (and the link at the very bottom) should we not move most of the CONTRIBUTING.md content to `iac/docs` and link to there to get everyone on the same page?

## Reviewing

> As a reviewer please go through the following checklist:

Reviewer: "confirm", "certify" ? text should make it clear that this is not optional.

- [ ] I understand what this PR is doing
- [ ] I can confirm it does what it is supposed to do
- [ ] The title is reflective of the changes
- [ ] The code readable
- [ ] Necessary tests have been added

> - [ ] Functionality documented/updated in the readme or [code](https://puppet.com/docs/puppet/latest/puppet_strings.html)

Reviewer: ```suggestion
- [ ] Functionality documented/updated in the readme and [REFERENCE](https://puppet.com/docs/puppet/latest/puppet_strings.html)
```

Do we have some documentation how/where REFERENCE.md gets updated? Is that part of the release-prep?
